cloudify_starlingx/utils.py

```diff
@@ -113,19 +114,34 @@ def update_openstack_props(ctx_instance, resources, client_config):
                                                     'os_password')))
 
 
-def assign_site(deployment_id, location):
-    site = get_site(deployment_id)
+def assign_site(ctx_instance, deployment_id, location):
+    config = ctx_instance.runtime_properties.get('resource_config', {})
+    location_name = format_location_name(config.get('location', ''))
+    site = get_site(location_name)
     if not site:
-        create_site(deployment_id, location)
+        create_site(location_name, location)
     elif not site.get('location'):
-        update_site(deployment_id, location)
-    update_deployment_site(deployment_id, deployment_id)
+        update_site(location_name, location)
+    update_deployment_site(deployment_id, location_name)
+
+
+def format_location_name(location_name):
+    return re.sub('\\-+', '-', re.sub('[^0-9a-zA-Z]', '-', location_name))
+
+
+def get_subcloud_group_id(ctx_instance):
+    subclouds = ctx_instance.runtime_properties.get('subcloud')
+    if subclouds:
+        for subcloud in subclouds.values():
+            return subcloud.get('group_id')
```

Comment:
What you return here? the group id of the first subcloud that has it?
can you add documentation to clear this?